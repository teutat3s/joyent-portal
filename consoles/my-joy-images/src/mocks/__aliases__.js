module.exports = {
  '^joyent-ui-toolkit/dist/es/editor$': '<rootDir>/src/mocks/editor',
  '^redux-form$': '<rootDir>/src/mocks/redux-form',
  '^react-responsive$': '<rootDir>/src/mocks/react-responsive',
  '^react-router-dom$': '<rootDir>/src/mocks/react-router-dom',
  '^declarative-redux-form$': '<rootDir>/src/mocks/declarative-redux-form',
  '^scroll-to-element': '<rootDir>/src/mocks/scroll-to-element'
};
