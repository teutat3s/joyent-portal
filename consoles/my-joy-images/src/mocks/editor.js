import React from 'react';

export default () => <span>joyent-maifest-editor</span>;
