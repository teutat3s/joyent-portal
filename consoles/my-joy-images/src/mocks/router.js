import React from 'react';
import { MemoryRouter } from 'react-router-dom';

export default ({ children }) => <MemoryRouter>{children}</MemoryRouter>;
