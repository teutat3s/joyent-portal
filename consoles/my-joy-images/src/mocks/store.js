import React from 'react';
import { ApolloProvider } from 'react-apollo';

import createClient from '@state/apollo-client';

export default ({ children }) => (
  <ApolloProvider client={createClient()}>{children}</ApolloProvider>
);
