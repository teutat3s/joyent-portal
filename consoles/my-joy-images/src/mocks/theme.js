import React from 'react';
import { ThemeProvider } from 'styled-components';
import {
  theme,
  RootContainer,
  PageContainer,
  ViewContainer
} from 'joyent-ui-toolkit';

export default ({ children, ss }) => (
  <ThemeProvider theme={theme}>
    {ss ? (
      <RootContainer>
        <PageContainer>
          <ViewContainer>{children}</ViewContainer>
        </PageContainer>
      </RootContainer>
    ) : (
      children
    )}
  </ThemeProvider>
);
