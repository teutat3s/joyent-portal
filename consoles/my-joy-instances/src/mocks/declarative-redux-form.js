import React from 'react';

export default ({ children, ...props }) => React.createElement(children, props);
