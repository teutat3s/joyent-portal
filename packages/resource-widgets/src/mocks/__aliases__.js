module.exports = {
  '^redux-form$': '<rootDir>/src/mocks/redux-form'
};
