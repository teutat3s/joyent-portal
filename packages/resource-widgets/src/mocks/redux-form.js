import React from 'react';

export const Field = ({ component = 'input', children, ...rest }) =>
  React.createElement(component, rest, children);
