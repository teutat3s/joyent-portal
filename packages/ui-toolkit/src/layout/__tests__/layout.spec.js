import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import remcalc from 'remcalc';

import { PageContainer, ViewContainer } from '../';
import { Theme } from '../../mocks';

describe('Layout', () => {
  test('PageContainer', () => {
    const tree = renderer
      .create(
        <Theme>
          <PageContainer />
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toHaveStyleRule('flex-flow', 'column');
    expect(tree).toHaveStyleRule('flex', '1 1 auto');
  });

  test('ViewContainer', () => {
    const tree = renderer
      .create(
        <Theme>
          <ViewContainer />
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('max-width', remcalc(1260));
  });

  test('ViewContainer fluid', () => {
    const tree = renderer
      .create(
        <Theme>
          <ViewContainer fluid />
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('padding-left', '0');
    expect(tree).toHaveStyleRule('padding-right', '0');
  });

  test('ViewContainer main', () => {
    const tree = renderer
      .create(
        <Theme>
          <ViewContainer main />
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('padding-bottom', remcalc(18));
  });

  test('ViewContainer center', () => {
    const tree = renderer
      .create(
        <Theme>
          <ViewContainer center />
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toHaveStyleRule('flex-direction', 'column');
    expect(tree).toHaveStyleRule('justify-content', 'center');
    expect(tree).toHaveStyleRule('align-content', 'center');
    expect(tree).toHaveStyleRule('align-items', 'center');
  });
});
