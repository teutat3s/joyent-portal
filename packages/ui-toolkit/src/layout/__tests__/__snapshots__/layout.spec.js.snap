// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout PageContainer 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  padding-bottom: 3.75rem;
}

<div
  className="c0"
/>
`;

exports[`Layout ViewContainer 1`] = `
.c1 {
  margin-right: auto;
  margin-left: auto;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  max-width: 78.75rem;
}

@media only screen and (min-width:48em) {
  .c1 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c1 {
    width: 56rem;
  }
}

@media only screen and (min-width:75em) {
  .c1 {
    width: 59rem;
  }
}

@media only screen and (max-width:37.4375rem) {
  .c0 {
    padding-left: 0.75rem;
    padding-right: 0.75rem;
  }
}

@media only screen and (min-width:37.5rem) and (max-width:62.4375rem) {
  .c0 {
    padding-left: 1.875rem;
    padding-right: 1.875rem;
  }
}

@media only screen and (min-width:62.5rem) and (max-width:87.4375rem) {
  .c0 {
    padding-left: 4.375rem;
    padding-right: 4.375rem;
  }
}

<div
  className="c0 c1"
/>
`;

exports[`Layout ViewContainer center 1`] = `
.c1 {
  margin-right: auto;
  margin-left: auto;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  max-width: 78.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media only screen and (min-width:48em) {
  .c1 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c1 {
    width: 56rem;
  }
}

@media only screen and (min-width:75em) {
  .c1 {
    width: 59rem;
  }
}

@media only screen and (max-width:37.4375rem) {
  .c0 {
    padding-left: 0.75rem;
    padding-right: 0.75rem;
  }
}

@media only screen and (min-width:37.5rem) and (max-width:62.4375rem) {
  .c0 {
    padding-left: 1.875rem;
    padding-right: 1.875rem;
  }
}

@media only screen and (min-width:62.5rem) and (max-width:87.4375rem) {
  .c0 {
    padding-left: 4.375rem;
    padding-right: 4.375rem;
  }
}

<div
  className="c0 c1"
/>
`;

exports[`Layout ViewContainer fluid 1`] = `
.c1 {
  margin-right: auto;
  margin-left: auto;
  padding-right: 1.875rem;
  padding-left: 1.875rem;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: 0;
  padding-right: 0;
}

<div
  className="c0 c1"
/>
`;

exports[`Layout ViewContainer main 1`] = `
.c1 {
  margin-right: auto;
  margin-left: auto;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  max-width: 78.75rem;
  padding-bottom: 1.125rem;
}

@media only screen and (min-width:48em) {
  .c1 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c1 {
    width: 56rem;
  }
}

@media only screen and (min-width:75em) {
  .c1 {
    width: 59rem;
  }
}

@media only screen and (max-width:37.4375rem) {
  .c0 {
    padding-left: 0.75rem;
    padding-right: 0.75rem;
  }
}

@media only screen and (min-width:37.5rem) and (max-width:62.4375rem) {
  .c0 {
    padding-left: 1.875rem;
    padding-right: 1.875rem;
  }
}

@media only screen and (min-width:62.5rem) and (max-width:87.4375rem) {
  .c0 {
    padding-left: 4.375rem;
    padding-right: 4.375rem;
  }
}

<div
  className="c0 c1"
/>
`;
